### ISSUE TITLE
Not warn CS8600 (may be not a bug)

### ISSUE THREAD

**Issue by Silencersn** [dotnet/roslyn#76551]
**Version Used**: 
VS 17.12.0
.NET 8

**Steps to Reproduce**:

```C#
object obj = default((object Obj, object Obj2)).Obj; // warns CS8600
object? obj2 = Array.Empty<(object Obj1, object Obj2)>().FirstOrDefault().Obj1;
object obj3 = obj2;  // not warn
```
(Is there any way to let it know that `obj3` now may be `null`, if it is not a bug.)

**Diagnostic Id**:
CS8600

**Expected Behavior**:
It warns CS8600.

**Actual Behavior**:
It does not warn.

**Comment by huoyaoyuan** [comment]
It's a well-known gap for nullability analysis because of the absence of struct-default analysis. There is special analysis for `default` literal, but the return values of `FirstOrDefault` etc are not considered as "may be default".
